src/pykeen/models/unimodal/boxe.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+
+"""Implementation of the BoxE model."""
+
+from typing import Any, ClassVar, Mapping, Optional
+
+from torch.nn.init import uniform_
+
+from ...constants import DEFAULT_EMBEDDING_HPO_EMBEDDING_DIM_RANGE
+from ...losses import NSSALoss
+from ...models import ERModel
+from ...nn.emb import EmbeddingSpecification
+from ...nn.init import uniform_norm_
+from ...nn.modules import BoxEInteraction
+from ...typing import Hint, Initializer
+
+__all__ = [
+    "BoxE",
+]
+
+
+class BoxE(ERModel):
+    r"""An implementation of BoxE from [abboud2020]_.
+
+    ---
+    citation:
+        author: Abboud
+        year: 2020
+        link: https://arxiv.org/abs/2007.06267
+        github: ralphabb/BoxE
+    """
+
+    #: The default strategy for optimizing the model's hyper-parameters
+    hpo_default: ClassVar[Mapping[str, Any]] = dict(
+        embedding_dim=DEFAULT_EMBEDDING_HPO_EMBEDDING_DIM_RANGE,
+        p=dict(type=int, low=1, high=2),
+    )
+
+    loss_default = NSSALoss
+    loss_default_kwargs = dict(margin=3, adversarial_temperature=2.0, reduction="sum")
+
+    def __init__(
+        self,
+        *,
+        embedding_dim: int = 200,
+        tanh_map: bool = True,
+        norm_order: int = 2,
+        entity_initializer: Hint[Initializer] = uniform_norm_,
+        entity_initializer_kwargs: Optional[Mapping[str, Any]] = None,
+        relation_initializer: Hint[Initializer] = uniform_norm_,  # Has to be scaled as well
+        relation_initializer_kwargs: Optional[Mapping[str, Any]] = None,
+        relation_size_initializer: Hint[Initializer] = uniform_,  # Has to be scaled as well
+        relation_size_initializer_kwargs: Optional[Mapping[str, Any]] = None,
+        **kwargs,
+    ) -> None:
+        r"""Initialize BoxE-KG.
+
+        :param embedding_dim:
+            The entity embedding dimension $d$. Defaults to 200. Is usually $d \in [50, 300]$.
+        :param tanh_map:
+            Whether to use tanh mapping after BoxE computation (defaults to true). The hyperbolic tangent mapping
+            restricts the embedding space to the range [-1, 1], and thus this map implicitly
+            regularizes the space to prevent loss reduction by growing boxes arbitrarily large.
+        :param norm_order:
+            Norm Order in score computation (Int): Default - 2
```

Comment:
```suggestion
            Norm Order in score computation
```
The default value & type are already part of the signature, so we do not need to repeat them.